packages/frontend/app/register/weekly-meetings/components/location-search.tsx

```diff
@@ -0,0 +1,160 @@
+                <FormItem>
+                  <FormLabel>Look for a location...</FormLabel>
+                  <FormDescription>Find a specific location</FormDescription>
+                  <FormControl>
+                    <Input
+                      placeholder="Street name, postcode or city"
+                      autoFocus
+                      {...field}
+                    />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          </form>
+        </Form>
+        <TooltipProvider>
+          <Tooltip>
+            <TooltipTrigger asChild>
+              <ShadButton
+                variant="outline"
+                onClick={() => {
+                  setUseCurrentLocation(true);
+                }}
+              >
+                <Locate />
+              </ShadButton>
+            </TooltipTrigger>
+            <TooltipContent>
+              <p>Use your current location</p>
+            </TooltipContent>
+          </Tooltip>
+        </TooltipProvider>
+      </div>
+
+      <ScrollArea className="max-h-[500px] h-[400px]">
+        {state.response && (
+          <motion.div
+            className="flex flex-col gap-y-2"
+            initial={{ opacity: 0 }}
+            animate={{ opacity: 1, transition: { ease: "easeIn" } }}
+            exit={{ opacity: 0 }}
+          >
+            {state.response.results.map(
+              (location: LocationSearchResult, i: number) => (
+                <Card key={i} className="h-[200px] min-h-[190px]">
+                  <CardHeader>
+                    <h2 className="font-bold">{location.formatted}</h2>
+                  </CardHeader>
+                  <CardBody>
+                    <div>
+                      <p>{location.city}</p>
+                      <p>{location.region}</p>
+                      <p>{location.country}</p>
+                    </div>
+                  </CardBody>
+                  <CardFooter>
+                    <Button
+                      variant="ghost"
+                      color="default"
+                      onClick={() => {
+                        // Fetch the meetings at the selected location
+                        dispatch(
+                          fetchMeetingsThunk({
+                            latitude: String(location.geometry.lat),
+                            longitude: String(location.geometry.lng),
+                          }),
+                        );
+                        dispatch(
+                          updateUserCoords({
+                            latitude: location.geometry.lat,
+                            longitude: location.geometry.lng,
+                          }),
+                        );
+                      }}
+                    >
+                      Select
+                    </Button>
+                  </CardFooter>
+                </Card>
+              ),
+            )}
+          </motion.div>
```

Comment:
This bit could also be refactored to be in a render fn (like we discussed before)